### PR TITLE
test/e2e: Allow zones to be passed to e2e cloud provider

### PR DESCRIPTION
**Pull request by smarterclayton**
Instead of allowing the cloud provider to guess at the zones that should be applied for a cluster under test, allow the explicit list
of zones to consider to be passed as a new test context flag -gce-zones.

For distributions that want to run e2e tests against the GCE cloud but restrict zones based on rules relevant to how they install, allowing explicit zone restrictions to be passed to the tests.

Only the GCE test cloud provider recognizes this value because only the GCE test cloud provider makes assumptions about zones for verifying values, and the default assumptions for GKE do not always match non-GKE providers.

/kind cleanup

```release-note
The new `-gce-zones` flag on the `e2e.test` binary instructs tests that check for information about how the cluster interacts with the cloud to limit their queries to the provided zone list. If not specified, the current behavior of asking the cloud provider for all available zones in multi zone clusters is preserved.
```

```docs

```

### CODE REVIEW

**Comment by smarterclayton** [comment]
/hold

(verifying there are no issues with this)

File: test/e2e/framework/providers/gce/gce.go [hunk 1]
```diff
@@ -79,7 +83,10 @@ func factory() (framework.ProviderInterface, error) {
 		return nil, fmt.Errorf("Error building GCE/GKE provider: %v", err)
 	}
 
-	// Arbitrarily pick one of the zones we have nodes in
+	// Arbitrarily pick one of the zones we have nodes in, looking at prepopulated zones first.
```
Comment (dims) [inline test/e2e/framework/providers/gce/gce.go:86]: Do we want to validate the "Must include Zone" restriction in the comment above here?

Comment (smarterclayton) [reply test/e2e/framework/providers/gce/gce.go:86]: Yeah, doing so

**Comment by dims** [comment]
/lgtm

please feel free to remove hold when you are ready

**Comment by smarterclayton** [comment]
/hold cancel

thanks for the review

**Pull request merged**